include/taskolib/Step.h

```diff
@@ -318,9 +341,10 @@ class Step
 
     /**
      * Execute the Lua script, throwing an exception if anything goes wrong.
-     * \see execute(Context&, CommChannel*, StepIndex)
+     * \see execute(Context&, CommChannel*, StepIndex, std::string&)
      */
-    bool execute_impl(Context& context, CommChannel* comm_channel, StepIndex index);
+    bool execute_impl(Context& context, CommChannel* comm_channel, StepIndex index,
+        std::string step_setup);
```

Comment:
As a general rule, we should prefer `const std::string&` for string parameters, as stated in the style guide. `gul14::string_view` is also good. But IMO a bare `std::string` or a `std::string&` needs some explanation.

Reply:
Changed to reference.

> ... But IMO a bare `std::string` or a `std::string&` needs some explanation.

Hmm, what do you mean with explanation? Do you mean the purpose of the variable `step_setup`? But if I describe `step_setup` then we also need to explain the others. I am little bit lost...

Reply:
Explanation why you did not use either `string const&` or `string_view`.

A reason to take a string as `std::string` is for example when you would copy it anyhow. If you want to avoid complicated code AND avoid unneeded copies taking a bare `std::string` is easiest, because you do not need to yourself check if the parameter is a temporary or not.

When you take a string as `string_view` and later assign it as `string_member_.assign(s.data(), s.length());` it becomes rather debatable if that is premature optimization and/or really more readable than the 'classic' approach.

I think ALL string optimization stuff should be checked for premature optimization.